src/replication/ng/scheduler/scheduler.go

```diff
@@ -42,3 +64,85 @@ type Scheduler interface {
 	// Stop the job specified by ID
 	Stop(id string) error
 }
+
+// Preprocess the resources and returns the item list that can be scheduled
+func (d *DefaultReplicator) Preprocess(srcResources []*model.Resource, destResources []*model.Resource) ([]*ScheduleItem, error) {
+	if len(srcResources) != len(destResources) {
+		err := errors.New("srcResources has different length with destResources")
```

Comment:
if this error will be bubbled to the users finally, I suggest we should use some more meaningful statements to describe the problem.